libbeat/docs/newbeat.asciidoc

```diff
@@ -146,203 +214,222 @@ NOTE: To be consistent with other Beats, you should append `beat` to your Beat n
-
-The `Setup` method enables you to execute logic before the main
-loop, usually for initialization. In the Topbeat implementation, this method
-only assigns the Beat object to the Topbeat object, so it doesn't have to be
-passed to all sub-functions.
-
-[source,go]
-----------------------------------------------------------------------
-func (tb *Topbeat) Setup(b *beat.Beat) error {
-
-	tb.Beat = b
-	return nil
-}
-----------------------------------------------------------------------
 
 [[run-method]]
 ==== Run Method
 
-The `Run` method should contain your main application loop. For Topbeat it looks
+The `Run` method should contain your main application loop. For `Lsbeat` it looks
 like this:
 
 [source,go]
 ----------------------------------------------------------------------
-func (t *Topbeat) Run(b *beat.Beat) error {
+func (bt *Lsbeat) Run(b *beat.Beat) error {
+	logp.Info("lsbeat is running! Hit CTRL-C to stop it.")
 
-	t.isAlive = true
+	bt.client = b.Publisher.Connect()
 
-	t.initProcStats()
+	bt.listDir(bt.config.Path, b.Name, true) // init directory
+	bt.lastIndexTime = time.Now()            // init indexing timestamp
 
-	var err error
+	ticker := time.NewTicker(bt.config.Period)
+	for {
 
-	for t.isAlive {
-		time.Sleep(t.period)
-
-		err = t.exportSystemStats()
-		if err != nil {
-			logp.Err("Error reading system stats: %v", err)
+		select {
+		case <-bt.done:
+			return nil
+		case <-ticker.C:
 		}
-		[...]
+
+		now := time.Now()
+		bt.listDir(bt.config.Path, b.Name, false) // call lsDir
+		bt.lastIndexTime = now                    // mark Timestamp
+
+		logp.Info("Event sent")
 	}
 
-	return err
 }
 ----------------------------------------------------------------------
 
-Inside the loop, Topbeat sleeps for a configurable period of time and then
-captures the required data and sends it to the publisher via the `events`
-publisher client. The publisher client is available as part of the Beat object
-through the `Beat.Events` variable.
+Inside the loop, `Lsbeat` sleeps for a configurable period of time and then
+captures the required data and sends it to the publisher. The publisher client is available as part of the Beat object
+through the `client` variable.
 
-The actual sending is done inside the `exportSystemStats()` method:
+The actual sending is done inside the `listDir()` method:
 
 [source,go]
 ----------------------------------------------------------------------
 
-func (t *Topbeat) exportSystemStats() error {
-
-	load_stat, err := GetSystemLoad()
-	if err != nil {
-		logp.Warn("Getting load statistics: %v", err)
-		return err
-	}
-
-	[...]
+func (bt *Lsbeat) listDir(dirFile string, beatname string, init bool) {
+	files, _ := ioutil.ReadDir(dirFile)
+	for _, f := range files {
+		t := f.ModTime()
+
+		event := common.MapStr{ <1>
+			"@timestamp": common.Time(time.Now()), <2>
+			"type":       beatname,
+			"modtime":    common.Time(t),
+			"filename":   f.Name(),
+			"path":       dirFile + "/" + f.Name(),
```

Comment:
`filepath.Join` will use system specific separator. `/` on unix and `\` on windows.